docs/pai-management/doc/how-to-add-new-conm.md

```diff
@@ -0,0 +1,200 @@
+## How to add new service configuration into cluster-object-model
+
+#### Index
+
+- [ Overveiw ](#Over)
+- [ Default Configuration ](#Default)
+- [ overwrite Configuration ](#Overwrite)
+- [ Service Parser ](#Service)
+- [ Document ](#Document_link)
+- [ How to write your template with cluster object model ](#use_link)
+
+
+#### Overview <a name="Over"></a>
+
+If you wanna add a new service configuration into cluster-object-model, you should do following things.
+
+- Design and write a default service configuration. 
+- Design and write overwrite service configuration for user to reconfigure the service in service-configuration.yaml
+- Design and write a service parser with python2.
+- Write a document to introduce your service object model.
+
+
+#### Default Configuration <a name="Default"></a>
+
+###### Path
+
+Please create the default configuration in the following path. 
+
+```bash
+src/${service_name}/config/${service_name}.yaml
+```
+
+
+###### Format
+
+${service_name}.yaml
+```yaml
+# key : value
+service-a-key1: default—value1
+
+service-a-key2: default-value2
+
+
+service-a-key3:
+    service-a-key3-subkey1: default-value3-subvalue1
+    service-a-key3-subkey2: default-value3-subvalue2
+    service-a-key3-subkey3: default-value3-subvalue3
+    
+service-a-key4:
+    - default-value4-sub1
+    - default-value4-sub2
+
+```
+
+#### overwrite Configuration <a name="Overwrite"></a>
+
+- Reconfigure the default value
+- Some properties is mandatory, user will have to configure it. Please put those configuration into the overwrite configuraiton in service-configuration.yaml
+
+###### Update example/cluster-configuration
+
+You should add the overwrite section in [example/service-configuration.yaml](../../../examples/cluster-configuration/services-configuration.yaml)
+
+Add the overwrite section of ```${service_name}``` like the following. Note, you could omit the default value which exists in default configuraiton.
+
+```yaml
+${service_name}:
+  
+  service-mandatory-key1: example-value1
+
+  service-a-key1: default—value1
+
+  service-a-key2: default-value2
+
+  service-a-key3:
+      service-a-key3-subkey1: default-value3-subvalue1
+      service-a-key3-subkey2: default-value3-subvalue2
+      service-a-key3-subkey3: default-value3-subvalue3
+    
+  service-a-key4:
+      - default-value4-sub1
+      - default-value4-sub2
+  
+``` 
+
+###### Update deployment/quick-start/services-configuration.yaml.template
+
+You should add the overwrite section in [deployment/quick-start/services-configuration.yaml.template](../../../deployment/quick-start/services-configuration.yaml.template)
+
+Add the overwrite section of ```${service_name}``` like the following. Please omit the default value, and only left the mandatory value.
+
+```yaml
+${service_name}:
+    service-mandatory-key1: usable-value1
+    
+```
+
+#### Service Parser <a name="Service"></a>
+
+###### path
+
+Please create the service.parser in the following path. 
```

Comment:
Is this mandatory ?

Reply:
Path?

Reply:
No, service.parser. If I didn't need parser, should I also create this?

Reply:
No. If you don't need parser, you could ignore this